### ISSUE TITLE
[3.0.0015] Unable to connect to the device using "Target Navigator"

### ISSUE THREAD

**Issue by pvallabhaneni** [blackberry/VSPlugin#702]
Environment: 
System OS:  Win 7 64bit and Win 8.1

Steps to Reproduce:

Scenario 1 on Non-Fresh Machine:
1. Connect a device to the system and upload a debug token from Blackberry->Options->signing page.
2. Goto Blackberry->Target Navigator and open Target Navigator Page.
3. Observe that the device connected is shown as connecting for longer time (more than 1 min).
4. Right click on the device and select "Disconnect"
5. Again Right click on the device and select "Connect"

Same issue is verified on another system.

Scenario 2 on Fresh Machine:
1. Connect a device to the system and upload a debug token from Blackberry->Options->signing page.
2. Goto Blackberry->Target Navigator and open Target Navigator Page.
3. Observe that the device is shown in connected state.
4. Right click on the device and select "Disconnect"
5. Again Right click on the device and select "Connect"

In both the cases, Following error is shown.

![capture](https://cloud.githubusercontent.com/assets/5992422/6186116/8e7a9b44-b39b-11e4-9f36-d8970763b5e8.PNG)

**Comment by phofman** [comment]
True. The public key is expected, but never generated unless you try to build your first project. Could be fixed.

**Comment by pvallabhaneni** [comment]
I have observed that "bbt_id_rsa_pub" file is not generated even if I build a project in any of the Scenarios and still device is not connecting.

Earlier In Scenario 2, Device got connected as  "bbt_id_rsa_pub" at "Research In Motion" is created from Momentics.

**Comment by phofman** [comment]
Sorry, you need to build & deploy. There is a dedicated step, that checks for `bbt_id_rsa.pub` existance and calls `ssh-keygen` if not.

**Comment by phofman** [comment]
Fixed as part of the #694 pull-request.

Now, each time connection to the device is about to be established, it will check for SSH public-key existance and create one, when not found. The only problem is, that it requires any NDK installed and selected, otherwise there is no `ssh-keygen.exe`.

This might potentially fix some other issues related to connectivity with missing SSH public key, like #679.
